dev-db/mongodb/mongodb-3.4.9.ebuild

```diff
@@ -0,0 +1,189 @@
+	sys-libs/readline
+	debug? ( dev-util/valgrind )
+	kerberos? ( dev-libs/cyrus-sasl[kerberos] )
+	test? (
+		dev-python/pymongo
+		dev-python/pyyaml
+	)"
+PDEPEND="tools? ( >=app-admin/mongo-tools-${PV} )"
+
+PATCHES=(
+	"${FILESDIR}/${PN}-3.2.10-boost-1.62.patch"
+	"${FILESDIR}/${PN}-3.4.0-fix-scons.patch"
+	"${FILESDIR}/${PN}-3.4.4-Replace-string-with-explicit-std-string.patch"
+	"${FILESDIR}/${PN}-3.4.6-sysmacros-include.patch"
+	"${FILESDIR}/${PN}-3.4.7-no-boost-check.patch"
+)
+
+S=${WORKDIR}/${MY_P}
+
+pkg_pretend() {
+	if [[ ${REPLACING_VERSIONS} < 3.0 ]]; then
+		ewarn "To upgrade from a version earlier than the 3.0-series, you must"
+		ewarn "successively upgrade major releases until you have upgraded"
+		ewarn "to 3.2-series. Then upgrade to 3.4 series."
+	fi
+}
+
+pkg_setup() {
+	enewgroup mongodb
+	enewuser mongodb -1 -1 /var/lib/${PN} mongodb
+
+	# Maintainer notes
+	#
+	# --use-system-tcmalloc is strongly NOT recommended:
+	# https://www.mongodb.org/about/contributors/tutorial/build-mongodb-from-source/
+
+	scons_opts=(
+		CC="$(tc-getCC)"
+		CXX="$(tc-getCXX)"
+
+		--disable-warnings-as-errors
+		--use-system-boost
+		--use-system-pcre
+		--use-system-snappy
+		--use-system-stemmer
+		--use-system-yaml
+		--use-system-zlib
+	)
+
+	# wiredtiger not supported on 32bit platforms #572166
+	use x86 && scons_opts+=( --wiredtiger=off )
+
+	if use debug; then
+		scons_opts+=( --dbg=on )
+	fi
+
+	if use prefix; then
+		scons_opts+=(
+			--cpppath="${EPREFIX}/usr/include"
+			--libpath="${EPREFIX}/usr/$(get_libdir)"
+		)
+	fi
+
+	if use kerberos; then
+		scons_opts+=( --use-sasl-client )
+	fi
+
+	if use ssl; then
+		scons_opts+=( --ssl )
+	fi
+}
+
+src_compile() {
+	# respect mongoDB upstream's basic recommendations
+	# see bug #536688 and #526114
+	if ! use debug; then
+		filter-flags '-m*'
+		filter-flags '-O?'
+	fi
+	escons "${scons_opts[@]}" core tools
+}
+
+src_install() {
+	escons "${scons_opts[@]}" --nostrip install --prefix="${ED}"/usr
+
+	local x
+	for x in /var/{lib,log}/${PN}; do
+		keepdir "${x}"
+		fowners mongodb:mongodb "${x}"
+	done
+
+	doman debian/mongo*.1
+	dodoc README docs/building.md
+
+	newinitd "${FILESDIR}/${PN}.initd-r2" ${PN}
+	newconfd "${FILESDIR}/${PN}.confd-r2" ${PN}
+	newinitd "${FILESDIR}/${PN/db/s}.initd-r2" ${PN/db/s}
+	newconfd "${FILESDIR}/${PN/db/s}.confd-r2" ${PN/db/s}
+
+	insinto /etc
+	newins "${FILESDIR}/${PN}.conf-r3" ${PN}.conf
+	newins "${FILESDIR}/${PN/db/s}.conf-r2" ${PN/db/s}.conf
+
+	systemd_dounit "${FILESDIR}/${PN}.service"
+
+	insinto /etc/logrotate.d/
+	newins "${FILESDIR}/${PN}.logrotate" ${PN}
+
+	# see bug #526114
+	pax-mark emr "${ED}"/usr/bin/{mongo,mongod,mongos}
+}
+
+pkg_preinst() {
+	# wrt bug #461466
+	if [[ "$(get_libdir)" == "lib64" ]]; then
+		rmdir "${ED}"/usr/lib/ &>/dev/null
+	fi
+}
+
+src_test() {
+	# this one test fails
+	rm jstests/core/jsHeapLimit.js || die
+
+	./buildscripts/resmoke.py --dbpathPrefix=test --suites core --jobs=$(makeopts_jobs) || die "Tests failed"
```

Comment:
I'm about to open a can of worms: could you add `${EPYTHON}` here before calling the `resmoke.py` script?

I realise this question is difficult to address because it means rewriting the ebuild including Python support. But if a Python interpreter is needed, we should technically make sure it's on the system before running tests.

Reply:
Ok, will do. Currently I added PYTHON_COMPAT and python-any-r1, but ${EPYTHON} doesn't seem to be set to anything. Will play with it later on.

Reply:
So after many trials and errors the problem is that mongodb has a pkg_config() that needed to be enriched by python-any-r1_pkg_setup().